slight improvement of autopilot

the-zorro opened the pull request:
- undock support
- stoping script on arriving to station

Viir commented:
Thanks for the PR, feel free to split it into multiple Commits/PRs as you see fit.

I see some issues with merging commit f68bfeaa6a1fe6ff4932664e77aaaa3ab56626de into master:

+ Visual Studio tells me the two versions of the file have different encodings as you can see the screenshot at [http://i.imgur.com/olXQFJy.png](http://i.imgur.com/olXQFJy.png) Can we keep UTF8 encoding?
+ I would prefer to keep around the simplest possible auto pilot as an example for other people who are new to programming (Hence the name "Travel.simple.cs"). Since your version takes into account more information and contains more branching I would prefer to offer it to the user in another script with a different file name.

Viir commented:
I do not understand why the control flow for `Undock` has been implemented this way:
Do we want to call `Undock` multiple times? If not, what is the reason that the call to `Undock` is contained in the loop block?

Why the `fly` variable? Can this not be simplified by breaking out of the loop as soon as the char is docked?

Viir commented on src/Sanderling/Sanderling.Exe/sample/script/Travel.simple.cs:
```diff
@@ -1,9 +1,40 @@
+			Host.Log("Arrived, stopping here");
+			goto loop;
+		} else {
+			Undock();
```
I do not understand why the control flow for `Undock` has been implemented this way:
Do we want to call `Undock` multiple times? If not, what is the reason that the call to `Undock` is contained in the loop block?

Viir commented on src/Sanderling/Sanderling.Exe/sample/script/Travel.simple.cs:
```diff
@@ -1,9 +1,40 @@
+	}
+}
 
+while(fly)
```
Why the `fly` variable? Can this not be simplified by using the condition used below to assign to `fly` as the looping condition?

the-zorro commented:
All true. Just focusing more on getting it work and getting familiar with
the API than keeping it pretty. But I'll change it tonight.
One more thing. I'm not from .Net world, so I'm using IntelliJ Rider as my
IDE and it doesn't assume some assemblies are referenced by default. Do you
mind if my changes will include some of those assemblies (System.Runtime,
System.IO) into project files?

On Dec 29, 2016 12:31 PM, "Michael Rätzel" <notifications@github.com> wrote:

> I do not understand why the control flow for Undock has been implemented
> this way:
> Do we want to call Undock multiple times? If not, what is the reason that
> the call to Undock is contained in the loop block?
>
> Why the fly variable? Can this not be simplified by breaking out of the
> loop as soon as the char is docked?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Arcitectus/Sanderling/pull/10#issuecomment-269688939>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABPKE_haq49x5Du5xLp76ROQ2-Ug07T8ks5rNBiQgaJpZM4LXW3p>
> .
>

Viir commented:
> I'm using IntelliJ Rider as my
> IDE and it doesn't assume some assemblies are referenced by default. Do you
> mind if my changes will include some of those assemblies (System.Runtime,
> System.IO) into project files?

If a change breaks results in Visual Studio being unable to build a project, than it is not okay as I assume most people use this. I do not see assemblies named System.IO or System.Runtime in the "Reference Manager" Dialog in VS. Are those part of the .NET framework?

the-zorro commented:
I added second commit. I removed Undock method entirely and just placed undock code before the main loop. Script stopping is done by break instead of variable. "simple" version is reverted, mine is called Travel.zorro.cs

the-zorro commented:
Interesting. Is there way to make it so that it will satisfy both tools?
The thing is not actually in Rider itself. Rider uses M$ build to build solutions. I'll check if the solution can be built by pure M$ build later today.

Viir commented:
> Interesting. Is there way to make it so that it will satisfy both tools?

I don't know.

the-zorro commented:
After dozen tries it seems fixed by installing one of M$ development packs. I think it was 4.6.1. Now both m$build and my IDE can compile unmodified projects.

Viir commented:
I see that you added a commit. Since commit #0d9bf3b is still in this PR, I assume it is not complete yet.

the-zorro commented:
It's not complete. I found one thing I'd like to change in the autopilot script, but that other commits you see are probably from another branch. In that branch I've modified mining script a lot, but I'm not sure it's a good idea to include it into your releases. Let me know if you think otherwise.

Viir commented:
The feature to stop the bot after docking is now implemented in the example autopilot bot at https://github.com/Viir/bots/commit/8b7ca9943601d5de1f1a6a68881ac76f9c4eae56

About the undocking feature: I am not sure this would be compatible with routes containing multiple stations: What would happen after the bot docks at the first station?

In any case, bots are now located in the dedicated repository, at https://github.com/Viir/bots
